server/src/main/java/org/eclipse/openvsx/publish/PublishExtensionVersionHandler.java

```diff
@@ -53,22 +60,23 @@ public class PublishExtensionVersionHandler {
     ExtensionValidator validator;
 
     @Transactional
-    public ExtensionVersion createExtensionVersion(ExtensionProcessor processor, PersonalAccessToken token) {
+    public ExtensionVersion createExtensionVersion(ExtensionProcessor processor, PersonalAccessToken token, LocalDateTime mirror) {
         // Extract extension metadata from its manifest
-        var extVersion = createExtensionVersion(processor, token.getUser(), token);
-        var dependencies = processor.getExtensionDependencies().stream()
-                .map(this::checkDependency)
-                .collect(Collectors.toList());
-        var bundledExtensions = processor.getBundledExtensions().stream()
-                .map(this::checkBundledExtension)
-                .collect(Collectors.toList());
-
-        extVersion.setDependencies(dependencies);
-        extVersion.setBundledExtensions(bundledExtensions);
+        var extVersion = createExtensionVersion(processor, token.getUser(), token, mirror);
+        var dependencies = processor.getExtensionDependencies().stream();
+        var bundledExtensions = processor.getBundledExtensions().stream();
+        if (mirror == null) {
+            // check dependencies, but skip in mirror mode
+            dependencies = dependencies.map(this::checkDependency);
+            bundledExtensions = bundledExtensions.map(this::checkBundledExtension);
+        }
+
+        extVersion.setDependencies(dependencies.collect(Collectors.toList()));
+        extVersion.setBundledExtensions(bundledExtensions.collect(Collectors.toList()));
         return extVersion;
     }
 
-    private ExtensionVersion createExtensionVersion(ExtensionProcessor processor, UserData user, PersonalAccessToken token) {
+    private ExtensionVersion createExtensionVersion(ExtensionProcessor processor, UserData user, PersonalAccessToken token, LocalDateTime mirror) {
```

Comment:
Please change `mirror` to `timestamp`.
Please replace the if-else `mirror != null` with `extVersion.setTimestamp(timestamp)`
When publish endpoint calls this method, pass `TimeUtil.getCurrentUTC()` as argument.